pinot-spi/src/main/java/org/apache/pinot/spi/config/table/UpsertConfig.java

```diff
@@ -154,10 +136,20 @@ public void setDefaultPartialUpsertStrategy(Strategy defaultPartialUpsertStrateg
    * same primary key, the record with the larger value of the time column is picked as the
    * latest update.
    * However, there are cases when users need to use another column to determine the order.
-   * In such case, you can use option comparisonColumn to override the column used for comparison.
+   * In such case, you can use option comparisonColumn to override the column used for comparison. When using
+   * multiple comparison columns, typically in the case of partial upserts, it is expected that input documents will
+   * each only have a singular non-null comparisonColumn. Multiple non-null values in an input document _will_ result
+   * in undefined behaviour. Typically, one comparisonColumn is allocated per distinct producer application of data
+   * in the case where there are multiple producers sinking to the same table.
    */
+  public void setComparisonColumns(List<String> comparisonColumns) {
```

Comment:
Just realize that certain configs might be handled unexpected. Consider the following configs:
```
{
  "comparisonColumn": "a",
  "comparisonColumns": null
}
```
```
{
  "comparisonColumn": null,
  "comparisonColumns": ["a"]
}
```
```
{
  "comparisonColumn": null
}
```
```
{
  "comparisonColumns": null
}
```

In order to handle all of them, we can modify these 2 methods in the following way:
```
  public void setComparisonColumns(List<String> comparisonColumns) {
    if (CollectionUtils.isNotEmpty(comparisonColumns) {
      _comparisonColumns = comparisonColumns;
    }
  }

  public void setComparisonColumn(String comparisonColumn) {
    if (comparisonColumn != null) {
      _comparisonColumns = Collections.singletonList(comparisonColumn);
    }
  }
```

Reply:
Good catch, made those suggested changes.